cclient/request/attach_to_tangle.c

```diff
@@ -12,39 +12,53 @@ attach_to_tangle_req_t* attach_to_tangle_req_new() {
       (attach_to_tangle_req_t*)malloc(sizeof(attach_to_tangle_req_t));
   if (req) {
     req->mwm = ATTACH_TO_TANGLE_MAIN_MWM;
-    req->trytes = flex_hash_array_new();
-    req->trunk = trit_array_new(NUM_TRITS_TRUNK);
-    req->branch = trit_array_new(NUM_TRITS_BRANCH);
+    req->trytes = NULL;
+    memset(req->trunk, FLEX_TRIT_NULL_VALUE, FLEX_TRIT_SIZE_243);
+    memset(req->branch, FLEX_TRIT_NULL_VALUE, FLEX_TRIT_SIZE_243);
   }
   return req;
 }
 
 void attach_to_tangle_req_free(attach_to_tangle_req_t** req) {
-  if (*req) {
-    trit_array_free((*req)->trunk);
-    trit_array_free((*req)->branch);
-    flex_hash_array_free((*req)->trytes);
-    free(*req);
-    *req = NULL;
+  if (!req || !(*req)) {
+    return;
   }
+
+  if ((*req)->trytes) {
+    hash8019_queue_free(&(*req)->trytes);
+  }
+
+  free(*req);
+  *req = NULL;
 }
 
 retcode_t attach_to_tangle_req_set_trunk(attach_to_tangle_req_t* req,
-                                         const char* trunk) {
-  return trytes_to_flex_hash(req->trunk, trunk);
+                                         tryte_t const* const trunk) {
+  size_t len = flex_trits_from_trytes(req->trunk, NUM_TRITS_HASH, trunk,
+                                      NUM_TRYTES_HASH, NUM_TRYTES_HASH);
+  return len ? RC_OK : RC_CCLIENT_FLEX_TRITS;
```

Comment:
You can go further and check that `len  == NUM_TRYTES_HASH `

Reply:
`flex_trits_from_trytes` returns `0` or a specific tryte number(depended on `num_trytes`) only, and `0` means error.

Reply:
My mistake sorry, I thought it could return a number different than `num_trytes`.